### ISSUE TITLE
Installation fails on new systems

### ISSUE THREAD

**Issue by GHoeher** [onetimesecret/onetimesecret#132]
Hi,

I tried to install onetimesecret on CentOS 8 as well as on a current Debian, but in all cases, I failed.

At the step "bin/ots init" I always get :
`/usr/bin/onetimesecret/lib/onetime/core_ext.rb:30: warning: constant ::Fixnum is deprecated
/usr/lib/ruby/gems/2.7.0/gems/storable-0.8.9/lib/proc_source.rb:41:in `<top (required)>': uninitialized constant RubyToken (NameError)
Did you mean?  RubyLex
        from /usr/lib/ruby/gems/2.7.0/gems/storable-0.8.9/lib/storable.rb:35:in `require'
        from /usr/lib/ruby/gems/2.7.0/gems/storable-0.8.9/lib/storable.rb:35:in `<class:Storable>'
        from /usr/lib/ruby/gems/2.7.0/gems/storable-0.8.9/lib/storable.rb:33:in `<top (required)>'
        from /usr/lib/ruby/gems/2.7.0/gems/sysinfo-0.7.3/lib/sysinfo.rb:2:in `require'
        from /usr/lib/ruby/gems/2.7.0/gems/sysinfo-0.7.3/lib/sysinfo.rb:2:in `<top (required)>'
        from /usr/bin/onetimesecret/lib/onetime.rb:12:in `require'
        from /usr/bin/onetimesecret/lib/onetime.rb:12:in `<top (required)>'
        from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        from bin/ots:9:in `<main>'`

I already tried to get a Ruby 1.9.x installed on CentOS 8 but there I failed for other reasons.

Any idea how I can get it up and running on a current CentOS (or Debian) with a current Ruby version ?

**Comment by Blackout8099** [comment]
Did you get anywhere with this?

**Comment by GHoeher** [comment]
Unfortunately not till now as I did not find the time to dig deeper.

**Comment by Blackout8099** [comment]
Fair enough... I tried on 7 same issue.

**Comment by hadesbox** [comment]
I have installation problems the gem install bundler steps fails like this

```
ubuntu@:~/sources/ruby-1.9.3-p362$ sudo gem install bundler
ERROR:  Error installing bundler:
        bundler requires Ruby version >= 2.3.0.
ubuntu@:~/sources/ruby-1.9.3-p362$
```

I installed ruby on this box following the steps on the installation guide for ubuntu
```
$ cd ~/sources
$ curl -O https://cache.ruby-lang.org/pub/ruby/1.9/ruby-1.9.3-p362.tar.bz2
$ tar xjf ruby-1.9.3-p362.tar.bz2
$ cd ruby-1.9.3-p362
$ ./configure && make
```

**Comment by jpablo365** [comment]
Was anyone able to successfully install OTS on centOS8?

**Comment by isAAAc** [comment]
on a fresh debian install: 
```
ots@myserver:/var/www/onetimesecret$ gem install bundler
Fetching: bundler-2.2.21.gem (100%)
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /var/lib/gems/2.5.0 directory.
```
isn't there some holes in the deploy process ?

**Comment by isAAAc** [comment]
https://github.com/onetimesecret/onetimesecret/issues/160  helped, 

but encountering a new issue`when lauching the service for the first time: 
```
ots@myserver:/var/www/onetimesecret$ bundle exec thin -d -S /var/run/thin/thin.sock -l /var/log/thin/thin.log -P /var/run/thin/thin.pid -e prod -s 2 restart
Traceback (most recent call last):
	20: from /var/www/onetimesecret/vendor/bundle/ruby/2.5.0/bin/thin:23:in `<main>'
	19: from /usr/lib/ruby/2.5.0/rubygems.rb:304:in `activate_bin_path'
	18: from /usr/lib/ruby/2.5.0/rubygems.rb:304:in `synchronize'
	17: from /usr/lib/ruby/2.5.0/rubygems.rb:306:in `block in activate_bin_path'
	16: from /usr/lib/ruby/2.5.0/rubygems.rb:243:in `finish_resolve'
	15: from /usr/lib/ruby/2.5.0/rubygems/request_set.rb:397:in `resolve_current'
	14: from /usr/lib/ruby/2.5.0/rubygems/request_set.rb:385:in `resolve'
	13: from /usr/lib/ruby/2.5.0/rubygems/resolver.rb:188:in `resolve'
	12: from /usr/lib/ruby/2.5.0/rubygems/resolver/molinillo/lib/molinillo/resolver.rb:42:in `resolve'
	11: from /usr/lib/ruby/2.5.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:64:in `resolve'
	10: from /usr/lib/ruby/2.5.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:106:in `start_resolution'
	 9: from /usr/lib/ruby/2.5.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:165:in `initial_state'
	 8: from /usr/lib/ruby/2.5.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:51:in `sort_dependencies'
	 7: from /usr/lib/ruby/2.5.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:69:in `with_no_such_dependency_error_handling'
	 6: from /usr/lib/ruby/2.5.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:52:in `block in sort_dependencies'
	 5: from /usr/lib/ruby/2.5.0/rubygems/resolver.rb:277:in `sort_dependencies'
	 4: from /usr/lib/ruby/2.5.0/rubygems/resolver.rb:277:in `with_index'
	 3: from /usr/lib/ruby/2.5.0/rubygems/resolver.rb:277:in `sort_by'
	 2: from /usr/lib/ruby/2.5.0/rubygems/resolver.rb:277:in `each'
	 1: from /usr/lib/ruby/2.5.0/rubygems/resolver.rb:283:in `block in sort_dependencies'
/usr/lib/ruby/2.5.0/rubygems/resolver.rb:231:in `search_for': Unable to resolve dependency: user requested 'did_you_mean (= 1.2.1)' (Gem::UnsatisfiableDependencyError)
```
any help is welcome ;)

**Comment by delano** [comment]
FYI, the original `Fixnum` error is fixed in [v0.11.0-RC1](https://github.com/onetimesecret/onetimesecret/tree/v0.11.0-RC1). It was related to a change between ruby 1.9-2.6 and 2.7. I need to do a bit more updating before pushing the gem, but you can clone it or switch your local copy to it:

```bash
$ git fetch 
$ git co -b v0.11.0-RC1 origin/v0.11.0-RC1
```

@GHoeher thanks for the initial report. Apologies for the extreme delay in responding. I was completely occupied with another project.

@hadesbox you can remove the bundler version line(s) in `Gemfile.lock` or update it with your version.

@isAAAc the fresh install error is a permissions error. You probably needed to use sudo. I've never seen that dependency error but I don't think anyone recommends running Ruby 2.5 so you're better off with 2.6.
